xdsl/frontend/onnx/context.py

```diff
@@ -0,0 +1,80 @@
+from onnx import GraphProto, NodeProto, ValueInfoProto
+
+from xdsl.builder import ImplicitBuilder
+from xdsl.dialects import func, onnx
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.frontend.onnx.type import get_type
+from xdsl.ir import Attribute, SSAValue
+from xdsl.irdl import IRDLOperation
+
+OP_BY_OP_TYPE: dict[str, type[IRDLOperation]] = {
+    "Add": onnx.Add,
+    "Sub": onnx.Sub,
+}
+"""Associate the name of the operations with the respective operation in ONNX dialect."""
+
+
+class OnnxXdslMapping:
+    """The representation of the onnx context."""
+
+    type_by_name: dict[str, Attribute]
+    value_by_name: dict[str, SSAValue]
+
+    def __init__(self):
+        self.type_by_name = {}
+        self.value_by_name = {}
+
+
+def visit_node(node: NodeProto, ctx: OnnxXdslMapping) -> None:
```

Comment:
can you please add a check that if we visit a node twice, it returns the existing xDSL operation in the ctx, and does not overwrite it with a new one?

Reply:
Similearly for `visit_value_info`